File: android/app/src/main/java/com/hatem/musicmute/SentryMonitoring.kt
package com.hatem.musicmute

import android.app.Application
import io.sentry.android.core.SentryAndroid

internal object SentryMonitoring {
    fun initialize(application: Application) {
        if (!BuildConfig.SENTRY_ENABLED || BuildConfig.SENTRY_DSN.isBlank()) return
        SentryAndroid.init(application) { options ->
            options.dsn = BuildConfig.SENTRY_DSN
            options.environment = "production"
            options.release = "musicmute-android@${BuildConfig.VERSION_NAME}"
            options.dist = "${BuildConfig.UPDATE_DISTRIBUTION}.${BuildConfig.VERSION_CODE}"
            options.maxBreadcrumbs = 0
            options.tracesSampleRate = 0.0
            options.isSendDefaultPii = false
            options.isAttachScreenshot = false
            options.isAttachViewHierarchy = false
            options.isEnableAutoSessionTracking = false
            options.beforeSend = io.sentry.SentryOptions.BeforeSendCallback { event, _ ->
                event.user = null
                event.request = null
                event.breadcrumbs = null
                event.message = null
                event.extras = emptyMap()
                event.contexts.clear()
                event.tags = mapOf("component" to "android")
                event.exceptions?.forEach { it.value = "Application failure" }
                event
            }
        }
    }
}

File: ios/Vocal/SentryMonitoring.swift
import Foundation
import Sentry

enum SentryMonitoring {
  static func start() {
    let bundle = Bundle.main
    guard bundle.object(forInfoDictionaryKey: "MusicMuteSentryEnabled") as? String == "YES",
      let dsn = bundle.object(forInfoDictionaryKey: "MusicMuteSentryDSN") as? String,
      let url = URL(string: dsn), url.scheme == "https", url.host != nil,
      url.user != nil
    else { return }

    SentrySDK.start { options in
      options.dsn = dsn
      options.environment = "production"
      let version =
        bundle.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? "unknown"
      let build = bundle.object(forInfoDictionaryKey: "CFBundleVersion") as? String ?? "unknown"
      options.releaseName = "musicmute-ios@\(version)"
      options.dist = build
      options.maxBreadcrumbs = 0
      options.tracesSampleRate = 0
      options.sendDefaultPii = false
      options.attachScreenshot = false
      options.attachViewHierarchy = false
      options.enableAutoSessionTracking = false
      options.beforeSend = { event in
        event.user = nil
        event.request = nil
        event.breadcrumbs = []
        event.extra = [:]
        event.context = [:]
        event.tags = ["component": "ios"]
        event.message = nil
        if let exceptions = event.exceptions {
          for exception in exceptions {
            exception.value = "Application failure"
          }
        }
        return event
      }
    }
  }
}
